### DISCUSSION TITLE
mockoon.cli in docker with YAML

### DISCUSSION THREAD

**Discussion by kockas** [mockoon/mockoon#1159] [Help] [answered]
Hi.

When I use mockoon cli directly, or I mount YAML file into existing mockoon image, everything works. When I put YAML file into image itself, it fails with following message:
`Error: This file is not a valid OpenAPI specification (JSON or YAML v2.0.0 and v3.0.0) or Mockoon environment: Unexpected token o in JSON at position 0`

Few comments:
As mentioned before. Directly in my shell this works:
`mockoon-cli start --data ~/mocked-api.openapi.yaml --port 3000`

mockoon-cli is of version 5.0.0 (but same behavior I had on version 3.

When I put the file into docker image using following Dockerfile
```
FROM mockoon/cli:5

ARG MOCKOON_FILE=
COPY --chown=mockoon $MOCKOON_FILE /mocked-api.openapi.yaml

EXPOSE 3000

CMD ["--data", "/mocked-api.openapi.yaml", "--port", "3000"]
```

The error above occurs.

Even if I jump into image:
`docker run -it --entrypoint /bin/sh mockoon-data`

And I run command manually from there, the result is same.
* There is no error in building image (I did SHA checksum)
* There is no access right issue, as in that case, mockoon cli would not complain, that it does not expect 'o' as first integer.
* Logs are empty
* Running mockoon/cli:5 for AMD64 platform has same result

I run it on Mac with M1 Pro (I guess, this it not an issue, but anyway mentioning it here).

Any hint, help, idea?

Thank you.

**Comment by 255kb** [comment]
Hi, 

Could you please share the file you are trying to run? 

I just tried with a similar setup, and a basic Mockoon data file and it worked. So, I suspect something is wrong with your file. 

Thank you

**Comment by 255kb** [comment, marked as answer]
After analyzing the file you sent me, could it be possible that you forgot to `COPY` the files that are referenced in your specification: $ref: "./file-name.openapi.yaml", etc. ?

The CLI is testing first if it's an OpenAPI spec file, if not it will try to parse as JSON. There is no error reporting on the first step. I will improve that for the next release.

**Reply by kockas** [reply]
You were right! That was it. Just message mentioning "o" as unexpected was really misleading, together with no logs nor any console output.

Regarding process, wouldn't you prefer to handle file based on suffix? Optionally define type (json/yaml) by some parameter in case, there is for some reason missing suffix? In that case ,you wouldn't get app complaining, that yaml starts with "o".

Thans once again.

**Reply by 255kb** [reply]
Yes, true we could add a check too to parse JSON only if it's a .json file. 
Issue is opened here: https://github.com/mockoon/mockoon/issues/1161
